contracts/main/price-feeders/DelayFathomOraclePriceFeed.sol

```diff
@@ -52,48 +50,88 @@ contract DelayFathomOraclePriceFeed is PausableUpgradeable, IFathomOraclePriceFe
+        return _isPriceOk();
+    }
+
+    function setAccessControlConfig(address _accessControlConfig) external onlyOwner {
+        IAccessControlConfig(_accessControlConfig).hasRole(IAccessControlConfig(_accessControlConfig).OWNER_ROLE(), msg.sender); // Sanity Check Call
+        accessControlConfig = IAccessControlConfig(_accessControlConfig);
+    }
+
     function setPriceLife(uint256 _second) external onlyOwner {
-        require(_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-price-life");
+        require(_second >= timeDelay && _second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-price-life");
         priceLife = _second;
         emit LogSetPriceLife(msg.sender, _second);
     }
 
     function setTimeDelay(uint256 _second) external onlyOwner {
-        require(_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-delay-time");
+        require(_second <= priceLife &&_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-delay-time");
+        _peekPrice();
         timeDelay = _second;
         emit LogSetTimeDelay(msg.sender, _second);
     }
 
+    function setToken0(address _token) external onlyOwner {
+        require(_token != address(0), "FathomOraclePriceFeed: ZERO_ADDRESS");
+        require(token1 != _token, "FathomOraclePriceFeed/same-token0-token1");
+
+        token0 = _token;
+    }
+
+    function setToken1(address _token) external onlyOwner {
+        require(_token != address(0), "FathomOraclePriceFeed: ZERO_ADDRESS");
+        require(token0 != _token, "FathomOraclePriceFeed/same-token0-token1");
+
+        token1 = _token;
+    }
+
+    function setOracle(address _oracle) external onlyOwner {
+        require(_oracle != address(0), "FathomOraclePriceFeed: ZERO_ADDRESS");
+        fathomOracle = IFathomOracle(_oracle);
+    }
+
     function pause() external onlyOwnerOrGov {
         _pause();
     }
 
     function unpause() external onlyOwnerOrGov {
         _unpause();
+        _peekPrice();
     }
 
-    function readPrice() external view override returns (bytes32) {
-        uint256 _price = currentPrice.price;
-        return bytes32(_price);
+    function setPrice() external onlyOwner {
+        _setPrice();
     }
 
-    function peekPrice() external override returns (bytes32, bool) {
-        if (currentPrice.price == 0 || block.timestamp >= currentPrice.lastUpdateTS + timeDelay) {
-            (uint256 _price, uint256 _lastUpdate) = fathomOracle.getPrice(token0, token1);
-            currentPrice.price = _price;
-            currentPrice.lastUpdateTS = _lastUpdate;
-            return (bytes32(currentPrice.price), _isPriceOk());
-        } else {
-            return (bytes32(currentPrice.price), _isPriceOk());
+    function _peekPrice() internal returns (bytes32, bool) {
+        if (block.timestamp >= lastUpdateTS + timeDelay) {
+          _setPrice();
         }
+        return (bytes32(delayedPrice), _isPriceOk());
```

Comment:
2.1.1 Current price is returned in the feed in DelayFathomOraclePriceFeed - fix, now peek price return delayed price instead of current one